### ISSUE TITLE
Endless waiting of XHR requests (Auto Waiter?)

### ISSUE THREAD

**Issue by mehmet-sap** [ui5-community/wdi5#569]
**Describe the bug**
WDI5 is indirectly testing a UI5 app that is continuesly sending XHR requests for updating a current status. This behavior is causing WDI5 to wait forever until the requests are resolved. In our case this will not be resolved so we cannot continue with testing an app. The test is running into a timeout.

Our workaround is to directly accessing the element via the id of for example a button. 

**To Reproduce**
Please contact me and I will show you that behavior in our application in a meeting.

**Expected behavior**
I expect to access all elements in the app via wdi5 selector. Currently no elements could be accessed only if we use  the following accessor: $(<id of your element which you want to access>).

**Logs/Console Output**
[0-0] [wdi5] call of _getControl() failed because of: Error: Polling stopped because the timeout of 60000 milliseconds has been reached but there is still pending asynchronous work.
[0-0] This is the last log of pending work:
[0-0] There are 1 open XHRs and 0 open FakeXHRs.
[0-0] XHR: URL: '/sap/opu/odata/sap/APS_EXT_ATO_IMP_SRV/$batch?sap-client=200' Method: 'POST' Async: 'true'
[0-0] Stack:     at Object.e [as resolveStackTrace] (<URL of our app>/resources/sap/ui/test/autowaiter/_utils.js?eval:6:117)
[0-0]     at XMLHttpRequest.d (<URL of our app>/resources/sap/ui/test/autowaiter/_XHRWaiter.js?eval:6:1318)
[0-0]     at XMLHttpRequest.send (<URL of our app>/resources/sap/ui/test/autowaiter/_XHRWaiter.js?eval:6:971)
[0-0]     at Object.request (<URL of our app>/resources/sap/ushell_abap/bootstrap/evo/abap.js:482:20511)
[0-0]     at Sr (<URL of our app>/resources/sap/ushell_abap/bootstrap/evo/abap.js:482:12217)
[0-0]     at n.request (<URL of our app>/resources/sap/ushell_abap/bootstrap/evo/abap.js:482:65316)
[0-0]     at z._request (<URL of our app>/resources/sap/ushell_abap/bootstrap/evo/core-ext-light-2.js:271:71732)
[0-0]     at g (<URL of our app>/resources/sap/ushell_abap/bootstrap/evo/core-ext-light-2.js:271:30467)
[0-0]     at <URL of our app>/resources/sap/ushell_abap/bootstrap/evo/core-ext-light-2.js:271:29893
[0-0]     at e (<URL of our app>/resources/sap/ui/test/autowaiter/_promiseWaiter.js?eval:6:2396) -  sap.ui.test.autowaiter._XHRWaiter#hasPending

**Runtime Env (please complete the following information):**

- `wdi5/wdio-ui5-service`-version: 1.5.6
- `UI5` version: 1.121.0
- `wdio`-version (output of `wdio --version`): 7.27.0
- `node`-version (output of `node --version`): v18.17.1
- OS: Windows 11
- Browser + Version: Chrome Version 120.0.6099.71 

**Additional context**
It would be very helpful to have an option to deactivate the auto wait feature.

**Comment by dominikfeininger** [comment]
All the waiting for the UI5 application is done inside the `RecordReplay` API.

Can @hmanchev provide some insides?

**Comment by Siolto** [comment]
We could probably add an optional `autoWait` property to the selector which is per default `true`. That would be the same as in [OPA5](https://sapui5.hana.ondemand.com/#/api/sap.ui.test.Opa5.BaseParameters). Something like that:

```javascript
browser.asControl({
  autoWait: false, // optional property - default: true
  selector: {
    // wdi5 selector...
  }
})
```

Further analysis what needs to be changed has to be done.

**Comment by hmanchev** [comment]
Siolto's idea seems reasonable for me.

**Comment by Siolto** [comment]
After further analyzing this issue we came to the conclusion that we don't want to add the possibility to disable the `autoWait`. There are just to many side effects and it would produce very fragile tests as a result. 

Instead we came up with the idea to give the possibility to exclude/ignore certain calls from the `autoWaiter`. This is already happening with specific OPA5 calls [here](https://github.com/SAP/openui5/blob/826f170b42216af3559cb4d6297008bdfecc0429/src/sap.ui.core/src/sap/ui/test/autowaiter/_XHRWaiter.js#L85).

@hmanchev would it be possible for your team to add a feature to the `autoWaiter`, that we could pass the information which calls should be excluded/ignored?
